### PR TITLE
ENG-9908 enable HTTPS on web/rest interface

### PR DESCRIPTION
**Pull request by manjujames**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java [hunk 1]
```diff
@@ -963,6 +977,67 @@ public HTTPAdminListener(
...
     }
 
+    private String getKeyTrustStoreAttribute(String sysPropName, KeyOrTrustStoreType store, String valueType, boolean throwForNull) {
+        String sysProp = System.getProperty(sysPropName);
+        if (StringUtils.isNotBlank(sysProp)) {
+            return sysProp.trim();
+        } else {
+            String value = null;
+            if (store!=null) {
+                value = valueType.equals("path") ? store.getPath() : store.getPassword();
```
Comment (vtkstef) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: `"path".equals(valueType)`

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java [hunk 2]
```diff
@@ -963,6 +977,67 @@ public HTTPAdminListener(
...
+            }
+            if (StringUtils.isBlank(value) && throwForNull) {
+                    throw new RuntimeException(
+                        "To enable HTTPS, keystore must be configured with password in deployment file or using system property");
```
Comment (vtkstef) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: IllegalArgumentException

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java [hunk 3]
```diff
@@ -963,6 +977,67 @@ public HTTPAdminListener(
...
+            } else {
+                return (value!=null) ? value.trim() : value;
```
Comment (vtkstef) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: are you trimming passwords? if so don't

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java [hunk 4]
```diff
@@ -963,6 +977,67 @@ public HTTPAdminListener(
...
+        if (value!=null) {
+            sslContextFactory.setTrustStorePath(value);
+        }
+        value = getKeyTrustStoreAttribute("javax.net.ssl.trustStorePassword", httpsType.getTrustStore(), "password", false);
+        if (value!=null) {
+            sslContextFactory.setTrustStorePassword(value);
+        }
+        /* More configurable things that we are not using for now.
+        sslContextFactory.setKeyManagerPassword("password");
+        sslContextFactory.setExcludeCipherSuites("SSL_RSA_WITH_DES_CBC_SHA",
```
Comment (vtkstef) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: I know these are weak ciphers so we should exclude them

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java (hunk 3 above)
Comment (manjujames) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: Good catch. I don't think there is a need to trim paths either, so I'll just get rid of the trim.

**Comment by akhanzode** [comment]
Does jetty automatically handle KEYSTORE_PASSWD_OBFUSCATED ? How does user specify this?

**Comment by manjujames** [comment]
Yes, jetty handles it automatically. In my test with obfuscated password, I document how to generate the obfuscated password.  Users will need to generate the obfuscated password and can use that instead of the clear text password.

**Comment by akhanzode** [comment]
Let Andrew know to document it.

**Pull request closed without merging**